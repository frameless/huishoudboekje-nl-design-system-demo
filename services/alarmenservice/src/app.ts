import cors from "cors";
import express from "express";
import apiV1Router from "./api-v1";
import {errorHandler} from "./errorHandlers";
import healthRouter from "./health";

const app = express();
const DATABASE_URL = process.env.DATABASE_URL;

if (!DATABASE_URL && process.env.NODE_ENV !== "test") {
	throw new Error("Unknown database. Please provide the database connection string in DATABASE_URL in the environment.");
}

const logger = () => (req, res, next) => {
	console.log("[EVENT]", `Incoming request: ${req.method} ${req.url}`);
	if (req.body) {
		console.log("[EVENT]", "Body:", req.body);
	}
	next();
};

app.use(cors());
app.use(express.json());
if (process.env.NODE_ENV === 'dev') {
	app.use(logger());
}

app.get("/health", healthRouter);
app.use("/v1/alarms", apiV1Router);
app.use(errorHandler);

export default app;