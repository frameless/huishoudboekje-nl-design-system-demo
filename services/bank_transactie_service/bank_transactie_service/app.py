""" Main app for bank_transactie_service """
import logging
import logging.config

from flask import Flask, Response
from flask_migrate import Migrate

from bank_transactie_service.views.bank_transaction import BankTransactionView
from bank_transactie_service.views.customer_statement_message import CustomerStatementMessageView
from core_service import database
from bank_transactie_service.views.bank_transaction_range import BanktransactionRangeView
from bank_transactie_service.views.transactions_filter_view import BanktransactionFilterView
from bank_transactie_service.views.bank_transaction_sum import BanktransactionSumView
from core_service.sqlalchemy_statsd_metrics import add_sqlalchemy_statsd_metrics

db = database.db


def create_app(config_name='bank_transactie_service.config.Config'):
    app = Flask(__name__)
    app.config.from_object(config_name)

    db.init_app(app)
    Migrate(app, db)

    logging.basicConfig(
        format='%(asctime)s %(levelname)-8s %(message)s',
        level=app.config["LOG_LEVEL"],
        datefmt='%Y-%m-%d %H:%M:%S')
    # logging.config.dictConfig(
    #     {
    #         "version": 1,
    #         "incremental": True,
    #         "loggers": {"sqlalchemy.engine": {"level": app.config["LOG_LEVEL"]}},
    #     }
    # )
    logging.info(f"Starting {__name__} with {config_name}")

    # Werkzeug has their own logger which outputs info level URL calls.
    # This can also cause parameters that are normally hidden to be logged
    logging.getLogger('werkzeug').setLevel(app.config["LOG_LEVEL"])

    add_sqlalchemy_statsd_metrics(app)


    @app.route('/health')
    def health():
        return Response()

    # Views
    routes = [
        {"path": "/customerstatementmessages", "view": CustomerStatementMessageView,
            "name": "customer_statement_message_view"},
        {"path": "/customerstatementmessages/<object_id>", "view": CustomerStatementMessageView,
            "name": "customer_statement_message_detail_view"},
        {"path": "/banktransactions", "view": BankTransactionView,
            "name": "banktransaction_view"},
        {"path": "/banktransactions/<object_id>", "view": BankTransactionView,
            "name": "banktransaction_detail_view"},
        {"path": "/banktransactions/filter", "view": BanktransactionFilterView,
            "name": "banktransaction_filter_view"},
        {"path": "/banktransactions/range", "view": BanktransactionRangeView,
            "name": "banktransaction_range_view"},
        {"path": "/banktransactions/sum", "view": BanktransactionSumView,
            "name": "banktransaction_sum_view"},
    ]
    for route in routes:
        app.add_url_rule(
            route["path"],
            view_func=route["view"].as_view(route["name"]),
            strict_slashes=False
        )
    return app


if __name__ == '__main__':
    create_app().run()
