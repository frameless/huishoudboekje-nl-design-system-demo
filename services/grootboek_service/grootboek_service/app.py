""" Main app for grootboek_service """
from models import *
import logging
from flask import Flask, Response
from flask_migrate import Migrate
from core_service import database
from grootboek_service.views.grootboekrekeningen import GrootboekrekeningenView
from core_service.statsd_metrics import add_statsd_metrics

db = database.db


def create_app(config_name='grootboek_service.config.Config'):
    app = Flask(__name__)
    app.config.from_object(config_name)

    logging.basicConfig(
        format='%(asctime)s %(levelname)-8s %(message)s',
        level=app.config["LOG_LEVEL"],
        datefmt='%Y-%m-%d %H:%M:%S')
    logging.info(f"Starting {__name__} with {config_name}")

    # Werkzeug has their own logger which outputs info level URL calls.
    # This can also cause parameters that are normally hidden to be logged
    logging.getLogger('werkzeug').setLevel(app.config["LOG_LEVEL"])

    db.init_app(app)
    Migrate(app, db)

    add_statsd_metrics(app)

    @app.route('/health')
    def health():
        return Response()

    # Views
    routes = [
        {"path": "/grootboekrekeningen", "view": GrootboekrekeningenView,
         "name": "grootboekrekening_view"},
        {"path": "/grootboekrekeningen/<object_id>", "view": GrootboekrekeningenView,
         "name": "grootboekrekening_detail_view"}
    ]
    for route in routes:
        app.add_url_rule(
            route["path"],
            view_func=route["view"].as_view(route["name"]),
            strict_slashes=False
        )
    return app


if __name__ == '__main__':
    create_app().run()
