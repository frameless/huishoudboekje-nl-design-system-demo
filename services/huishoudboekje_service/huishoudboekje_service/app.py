""" Main app for huishoudboekje_service """
import logging
import os
from flask import Flask, Response
from flask_migrate import Migrate

from huishoudboekje_service.views import (
    BurgerView,
    AfspraakView,
    RekeningView,
    RekeningBurgerView,
    RekeningAfdelingView,
    JournaalpostView,
    RubriekView,
    ConfiguratieView,
    OverschrijvingView,
    ExportView,
    HuishoudenView,
    AfdelingView,
)
from core_service import database

db = database.db


def create_app(config_name='huishoudboekje_service.config.Config'):
    app = Flask(__name__)
    app.config.from_object(config_name)

    logging.basicConfig(
        format='%(asctime)s %(levelname)-8s %(message)s',
        level=app.config["LOG_LEVEL"],
        datefmt='%Y-%m-%d %H:%M:%S')
    logging.info(f"Starting {__name__} with {config_name}")

    db.init_app(app)
    migrate = Migrate(app, db)

    @app.route('/health')
    def health():
        return Response()

    # Views
    routes = [
        {"path": "/burgers", "view": BurgerView, "name": "burger_view"},
        {"path": "/burgers/<object_id>", "view": BurgerView, "name": "burger_detail_view"},
        {"path": "/burgers/<object_id>/rekeningen", "view": RekeningBurgerView,
         "name": "burger_rekeningen_view"},
        {"path": "/afspraken", "view": AfspraakView, "name": "afspraak_view"},
        {"path": "/afspraken/<object_id>", "view": AfspraakView, "name": "afspraak_detail_view"},
        {"path": "/rekeningen", "view": RekeningView, "name": "rekening_view"},
        {"path": "/rekeningen/<object_id>", "view": RekeningView, "name": "rekening_detail_view"},
        {"path": "/journaalposten", "view": JournaalpostView, "name": "journaalpost_view"},
        {"path": "/journaalposten/<object_id>", "view": JournaalpostView, "name": "journaalpost_detail_view"},
        {"path": "/rubrieken", "view": RubriekView, "name": "rubriek_view"},
        {"path": "/rubrieken/<object_id>", "view": RubriekView, "name": "rubriek_detail_view"},
        {"path": "/configuratie", "view": ConfiguratieView, "name": "configuratie_view"},
        {"path": "/configuratie/<object_id>", "view": ConfiguratieView, "name": "configuratie_detail_view"},
        {"path": "/overschrijvingen", "view": OverschrijvingView, "name": "overschrijving_view"},
        {"path": "/overschrijvingen/<object_id>", "view": OverschrijvingView, "name": "overschrijving_detail_view"},
        {"path": "/export", "view": ExportView, "name": "export_view"},
        {"path": "/export/<object_id>", "view": ExportView, "name": "export_detail_view"},
        {"path": "/huishoudens", "view": HuishoudenView, "name": "huishouden_view"},
        {"path": "/huishoudens/<object_id>", "view": HuishoudenView, "name": "huishouden_detail_view"},
        {"path": "/afdelingen", "view": AfdelingView, "name": "afdeling_view"},
        {"path": "/afdelingen/<object_id>", "view": AfdelingView, "name": "afdeling_detail_view"},
        {"path": "/afdelingen/<object_id>/rekeningen", "view": RekeningAfdelingView,
         "name": "afdeling_rekeningen_view"},
    ]
    for route in routes:
        app.add_url_rule(
            route["path"],
            view_func=route["view"].as_view(route["name"]),
            strict_slashes=False
        )
    return app
