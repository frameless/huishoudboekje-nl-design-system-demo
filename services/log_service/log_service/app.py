""" Main app for log_service """
import logging
import logging.config

from flask import Flask, Response
from flask_migrate import Migrate

from core_service import database
from log_service.views.gebruikersactiviteit import GebruikersActiviteitView
from core_service.statsd_metrics import add_statsd_metrics

db = database.db


def create_app(
    config_name="log_service.config.Config"
):
    app = Flask(__name__)
    app.config.from_object(config_name)

    db.init_app(app)
    Migrate(app, db)

    logging.basicConfig(
        format='%(asctime)s %(levelname)-8s %(message)s',
        level=app.config["LOG_LEVEL"],
        datefmt='%Y-%m-%d %H:%M:%S')
    # logging.config.dictConfig(
    #     {
    #         "version": 1,
    #         "incremental": True,
    #         "loggers": {"sqlalchemy.engine": {"level": app.config["LOG_LEVEL"]}},
    #     }
    # )
    logging.info(f"Starting {__name__} with {config_name}")

    # Werkzeug has their own logger which outputs info level URL calls.
    # This can also cause parameters that are normally hidden to be logged
    logging.getLogger('werkzeug').setLevel("WARNING")

    add_statsd_metrics(app)

    @app.route("/health")
    def health():
        return Response()

    # Views
    routes = [
        {
            "path": "/gebruikersactiviteiten",
            "view": GebruikersActiviteitView,
            "name": "gebruikersactiviteit_view",
        },
        {
            "path": "/gebruikersactiviteiten/<object_id>",
            "view": GebruikersActiviteitView,
            "name": "gebruikersactiviteit_detail_view",
        },
    ]
    for route in routes:
        app.add_url_rule(
            route["path"],
            view_func=route["view"].as_view(route["name"]),
            strict_slashes=False,
        )

    return app


if __name__ == "__main__":
    create_app().run()
