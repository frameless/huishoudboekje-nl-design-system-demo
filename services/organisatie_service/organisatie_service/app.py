""" Main app for organisatie_service """
from models import *
import logging

from flask import Flask, Response
from flask_migrate import Migrate

from organisatie_service.views.organisaties import OrganisatieView
from organisatie_service.views.organisatie_rekeningen import OrganisatieRekeningenView
from organisatie_service.views.afdelingen import AfdelingView
from core_service import database
from core_service.sqlalchemy_statsd_metrics import add_sqlalchemy_statsd_metrics

db = database.db


def create_app(config_name='organisatie_service.config.Config'):
    app = Flask(__name__)
    app.config.from_object(config_name)

    logging.basicConfig(
        format='%(asctime)s %(levelname)-8s %(message)s',
        level=app.config["LOG_LEVEL"],
        datefmt='%Y-%m-%d %H:%M:%S')
    logging.info(f"Starting {__name__} with {config_name}")

    # Werkzeug has their own logger which outputs info level URL calls.
    # This can also cause parameters that are normally hidden to be logged
    logging.getLogger('werkzeug').setLevel(app.config["LOG_LEVEL"])

    db.init_app(app)
    Migrate(app, db)

    add_sqlalchemy_statsd_metrics(app)

    @app.route('/health')
    def health():
        return Response()

    routes = [
        {"path": "/organisaties", "view": OrganisatieView, "name": "organisatie_view"},
        {"path": "/organisaties/<object_id>", "view": OrganisatieView,
            "name": "organisatie_detail_view"},
        {"path": "/organisaties/rekeningen", "view": OrganisatieRekeningenView, "name": "organisatie_rekeningen_view"},
        {"path": "/afdelingen", "view": AfdelingView, "name": "afdeling_view"},
        {"path": "/afdelingen/<object_id>", "view": AfdelingView,
            "name": "afdeling_detail_view"},
    ]

    for route in routes:
        app.add_url_rule(
            route["path"],
            view_func=route["view"].as_view(route["name"]),
            strict_slashes=False
        )
    return app


if __name__ == '__main__':
    create_app().run()
