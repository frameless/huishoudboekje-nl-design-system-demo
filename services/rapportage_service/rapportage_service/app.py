""" Main app for rapportage_service """
import logging
import logging.config

from flask import Flask, Response
from flask_injector import FlaskInjector
from rapportage_service.dependencies import configure
from rapportage_service.views.RapportageView import BurgerRapportageView


def create_app(config_name='rapportage_service.config.Config', dependency_injection_configuration=configure):
    app = Flask(__name__)
    app.config.from_object(config_name)

    logging.basicConfig(
        format='%(asctime)s %(levelname)-8s %(message)s',
        level=app.config["LOG_LEVEL"],
        datefmt='%Y-%m-%d %H:%M:%S')
    logging.config.dictConfig(
        {
            "version": 1,
            "incremental": True,
            "loggers": {"sqlalchemy.engine": {"level": app.config["LOG_LEVEL"]}},
        }
    )
    logging.info(f"Starting {__name__} with {config_name}")

    # Werkzeug has their own logger which outputs info level URL calls.
    # This can also cause parameters that are normally hidden to be logged
    logging.getLogger('werkzeug').setLevel(app.config["LOG_LEVEL"])

    @app.route('/health')
    def health():
        return Response()

    # Views
    routes = [
        {"path": "/rapportage", "view": BurgerRapportageView,
         "name": "rapportage_burger"}
    ]
    for route in routes:
        app.add_url_rule(
            route["path"],
            view_func=route["view"].as_view(route["name"]),
            strict_slashes=False
        )

    # Initialize Flask-Injector. This needs to be run after you attached all
    # views, handlers, context processors and template globals.
    FlaskInjector(app=app, modules=[dependency_injection_configuration])
    return app


if __name__ == '__main__':
    create_app().run()
