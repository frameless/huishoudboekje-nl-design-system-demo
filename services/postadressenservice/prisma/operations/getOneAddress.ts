import prisma from "../../src/client";
import {NotFoundError} from "../../src/errorHandlers";

const getOneAddress = async (id: string) => {
	const data = await prisma.address.findUnique({
		where: {id},
	});

	if (!data) {
		throw new NotFoundError();
	}

	return data;
};

export default getOneAddress;