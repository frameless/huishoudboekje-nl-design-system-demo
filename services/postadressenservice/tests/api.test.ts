import {Prisma} from "@prisma/client";
import supertest from "supertest";
import {prismaMock} from "../.jest/mockClient";
import getOneAddress from "../prisma/operations/getOneAddress";
import app from "../src/app";
import addresses from "./fixtures/addresses";
import ResolvedValue = jest.ResolvedValue;

const api = supertest(app);

describe("Address CRUD (operations)", () => {

	describe("Given the address exists", () => {
		it("should return the address", async () => {
			prismaMock.address.findUnique.mockResolvedValue(addresses[0]);

			const result = await api.get("/v1/addresses/" + addresses[0].id);
			expect(result.statusCode).toBe(200);
			expect(result.body).toMatchSnapshot();
		});

		it("should return all addresses", async () => {
			prismaMock.address.findMany.mockResolvedValue(addresses as ResolvedValue<unknown>);

			const result = await api.get("/v1/addresses/");
			expect(result.statusCode).toBe(200);
			expect(result.body).toMatchSnapshot();
		});

		it("should return addresses filtered by id (through request query params)", async () => {
			const filterIds = addresses.map(a => a.id).slice(0, 2);
			prismaMock.address.findMany.mockResolvedValue(addresses.filter(a => filterIds.includes(a.id)) as ResolvedValue<unknown>);

			const result = await api.get("/v1/addresses/?filter_ids=" + filterIds.join(","));
			expect(result.statusCode).toBe(200);
			expect(result.body).toMatchSnapshot();
		});

		it("should return addresses filtered by id (through request body)", async () => {
			const filterIds = addresses.map(a => a.id).slice(0, 2);
			prismaMock.address.findMany.mockResolvedValue(addresses.filter(a => filterIds.includes(a.id)) as ResolvedValue<unknown>);

			const result = await api.get("/v1/addresses/").send({
				filter_ids: filterIds,
			});
			expect(result.statusCode).toBe(200);
			expect(result.body).toMatchSnapshot();
		});

		it("should update the street", async () => {
			const newStreet = "Barstreet";
			prismaMock.address.update.mockResolvedValue({
				...addresses[0],
				street: newStreet,
			} as ResolvedValue<unknown>);

			const result = await api.put("/v1/addresses/" + addresses[0].id).send({
				street: newStreet,
			});
			expect(result.statusCode).toBe(200);
			expect(result.body).toMatchSnapshot();
		});

		it("should update the house number", async () => {
			const newHouseNumber = "123AB";
			prismaMock.address.update.mockResolvedValue({
				...addresses[0],
				houseNumber: newHouseNumber,
			} as ResolvedValue<unknown>);

			const result = await api.put("/v1/addresses/" + addresses[0].id).send({
				houseNumber: newHouseNumber,
			});
			expect(result.statusCode).toBe(200);
			expect(result.body).toMatchSnapshot();
		});

		it("should update the postalcode", async () => {
			const newPostalCode = "1111AA";
			prismaMock.address.update.mockResolvedValue({
				...addresses[0],
				postalCode: newPostalCode,
			} as ResolvedValue<unknown>);

			const result = await api.put("/v1/addresses/" + addresses[0].id).send({
				postalCode: newPostalCode,
			});
			expect(result.statusCode).toBe(200);
			expect(result.body).toMatchSnapshot();
		});

		it("should update the locality", async () => {
			const newLocality = "Diemen";
			prismaMock.address.update.mockResolvedValue({
				...(addresses[0]),
				locality: newLocality,
			} as ResolvedValue<unknown>);

			const result = await api.put("/v1/addresses/" + addresses[0].id).send({
				locality: newLocality,
			});
			expect(result.statusCode).toBe(200);
			expect(result.body).toMatchSnapshot();
		});

		it("should update the Address", async () => {
			const newAddress = {
				street: "Bazstreet",
				houseNumber: "99",
				postalCode: "4444RR",
				locality: "Utrecht",
			};
			const timeCreated = addresses[0].timeCreated;

			prismaMock.address.update.mockResolvedValue({
				...newAddress,
				id: addresses[0].id,
				timeCreated,
			} as ResolvedValue<unknown>);

			const result = await api.put("/v1/addresses/" + addresses[0].id).send({
				...newAddress,
			});
			expect(result.statusCode).toBe(200);
			expect(result.body).toMatchSnapshot();
		});

		it("should delete the Address", async () => {
			prismaMock.address.delete.mockResolvedValue(undefined!);

			const result = await api.delete("/v1/addresses/" + addresses[0].id);
			expect(result.statusCode).toBe(204);
		});
	});

	describe("Given the address does not exist", () => {
		it("should return 404 Not Found when tyring to getOneAddress", async () => {
			prismaMock.address.update.mockRejectedValue(new Prisma.PrismaClientKnownRequestError("Not found", {code: "P2025", clientVersion: "0.0.0"}));

			const result = await api.get("/v1/addresses/" + addresses[0].id);
			expect(result.statusCode).toEqual(404);
		});

		it("should create the address", async () => {
			prismaMock.address.create.mockResolvedValue(addresses[0]);

			const result = await api.post("/v1/addresses").send(addresses[0]);
			expect(result.statusCode).toBe(201);
			expect(result.body).toMatchSnapshot();
		});

		it("should return 404 Not Found when trying to updateAddress", async () => {
			const newStreet = "Barstreet";
			prismaMock.address.update.mockRejectedValue(new Prisma.PrismaClientKnownRequestError("Not found", {code: "P2025", clientVersion: "0.0.0"}));

			const result = await api.put("/v1/addresses/" + addresses[0].id).send({
				id: addresses[0].id,
				street: newStreet,
			});
			expect(result.statusCode).toEqual(404);
		});

		it("should delete the Address", async () => {
			prismaMock.address.delete.mockRejectedValue(new Prisma.PrismaClientKnownRequestError("Not found", {code: "P2025", clientVersion: "0.0.0"}));
			const result = await api.delete("/v1/addresses/" + addresses[0].id);
			expect(result.statusCode).toEqual(204);
		});
	});

	describe("Given an unknown error occurs on the server", () => {
		const error = new Error("Server did a boo boo");

		it("should return a 500 when tyring to getOneAddress", async () => {
			prismaMock.address.findUnique.mockRejectedValue(error);
			const getAddressResult = getOneAddress(addresses[0].id);
			await expect(getAddressResult).rejects.toEqual(error);

			const result = await api.get("/v1/addresses/" + addresses[0].id);
			expect(result.statusCode).toEqual(500);
		});

		it("should return a 500 when tyring to getManyAddresses", async () => {
			prismaMock.address.findMany.mockRejectedValue(error);
			const ids = addresses.map(a => a.id);

			const result = await api.get("/v1/addresses/?filter_ids=" + ids.join(","));
			expect(result.statusCode).toEqual(500);
		});

		it("should return a 500 when trying to create the address", async () => {
			prismaMock.address.create.mockRejectedValue(error);

			const result = await api.post("/v1/addresses").send(addresses[0]);
			expect(result.statusCode).toBe(500);
			expect(result.body).toMatchSnapshot();
		});

		it("should return a 500 when trying to updateAddress", async () => {
			prismaMock.address.update.mockRejectedValue(error);

			const newStreet = "Barstreet";
			const result = await api.put("/v1/addresses/" + addresses[0].id).send({
				id: addresses[0].id,
				street: newStreet,
			});
			expect(result.statusCode).toEqual(500);
		});

		it("should return a 500 when trying to deleteAddress", async () => {
			prismaMock.address.delete.mockRejectedValue(error);
			const result = await api.delete("/v1/addresses/" + addresses[0].id);
			expect(result.statusCode).toEqual(500);
		});
	});

	describe("Given the server is healthy", () => {
		it("should return 200 OK", async () => {
			const result = await api.get("/health");
			expect(result.statusCode).toEqual(200);
		});
	});

});
