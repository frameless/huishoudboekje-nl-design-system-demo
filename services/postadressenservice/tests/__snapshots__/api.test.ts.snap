// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Address CRUD (operations) Given an unknown error occurs on the server should return a 500 when trying to create the address 1`] = `
{
  "message": "Server did a boo boo",
  "ok": false,
}
`;

exports[`Address CRUD (operations) Given the address does not exist should create the address 1`] = `
{
  "data": {
    "houseNumber": "1",
    "id": "3fc9baf4-f85b-4643-b525-bb901ab825dc",
    "locality": "Sloothuizen",
    "postalCode": "9999ZZ",
    "street": "Foostreet",
    "timeCreated": "2022-05-10T12:00:00.000Z",
  },
  "ok": true,
}
`;

exports[`Address CRUD (operations) Given the address exists should return all addresses 1`] = `
{
  "data": [
    {
      "houseNumber": "1",
      "id": "3fc9baf4-f85b-4643-b525-bb901ab825dc",
      "locality": "Sloothuizen",
      "postalCode": "9999ZZ",
      "street": "Foostreet",
      "timeCreated": "2022-05-10T12:00:00.000Z",
    },
    {
      "houseNumber": "1",
      "id": "14769fb4-46be-4a87-84d1-da49833ca786",
      "locality": "Sloothuizen",
      "postalCode": "9999ZZ",
      "street": "Foostreet",
      "timeCreated": "2022-05-10T12:00:00.000Z",
    },
    {
      "houseNumber": "1",
      "id": "7c8a52e5-0ed4-4909-a042-6273c537e4af",
      "locality": "Sloothuizen",
      "postalCode": "9999ZZ",
      "street": "Foostreet",
      "timeCreated": "2022-05-10T12:00:00.000Z",
    },
    {
      "houseNumber": "1",
      "id": "da3a73e8-826c-426a-9aff-4f64d01d88c4",
      "locality": "Sloothuizen",
      "postalCode": "9999ZZ",
      "street": "Foostreet",
      "timeCreated": "2022-05-10T12:00:00.000Z",
    },
  ],
  "ok": true,
}
`;

exports[`Address CRUD (operations) Given the address exists should return the address 1`] = `
{
  "data": {
    "houseNumber": "1",
    "id": "3fc9baf4-f85b-4643-b525-bb901ab825dc",
    "locality": "Sloothuizen",
    "postalCode": "9999ZZ",
    "street": "Foostreet",
    "timeCreated": "2022-05-10T12:00:00.000Z",
  },
  "ok": true,
}
`;

exports[`Address CRUD (operations) Given the address exists should update the Address 1`] = `
{
  "data": {
    "houseNumber": "99",
    "id": "3fc9baf4-f85b-4643-b525-bb901ab825dc",
    "locality": "Utrecht",
    "postalCode": "4444RR",
    "street": "Bazstreet",
    "timeCreated": "2022-05-10T12:00:00.000Z",
  },
  "ok": true,
}
`;

exports[`Address CRUD (operations) Given the address exists should update the house number 1`] = `
{
  "data": {
    "houseNumber": "123AB",
    "id": "3fc9baf4-f85b-4643-b525-bb901ab825dc",
    "locality": "Sloothuizen",
    "postalCode": "9999ZZ",
    "street": "Foostreet",
    "timeCreated": "2022-05-10T12:00:00.000Z",
  },
  "ok": true,
}
`;

exports[`Address CRUD (operations) Given the address exists should update the locality 1`] = `
{
  "data": {
    "houseNumber": "1",
    "id": "3fc9baf4-f85b-4643-b525-bb901ab825dc",
    "locality": "Diemen",
    "postalCode": "9999ZZ",
    "street": "Foostreet",
    "timeCreated": "2022-05-10T12:00:00.000Z",
  },
  "ok": true,
}
`;

exports[`Address CRUD (operations) Given the address exists should update the postalcode 1`] = `
{
  "data": {
    "houseNumber": "1",
    "id": "3fc9baf4-f85b-4643-b525-bb901ab825dc",
    "locality": "Sloothuizen",
    "postalCode": "1111AA",
    "street": "Foostreet",
    "timeCreated": "2022-05-10T12:00:00.000Z",
  },
  "ok": true,
}
`;

exports[`Address CRUD (operations) Given the address exists should update the street 1`] = `
{
  "data": {
    "houseNumber": "1",
    "id": "3fc9baf4-f85b-4643-b525-bb901ab825dc",
    "locality": "Sloothuizen",
    "postalCode": "9999ZZ",
    "street": "Barstreet",
    "timeCreated": "2022-05-10T12:00:00.000Z",
  },
  "ok": true,
}
`;
